### ISSUE TITLE
Send found lines to a new file window

### ISSUE THREAD

**Issue by zbika73** [notepad-plus-plus/notepad-plus-plus#11364]
### Send found lines to a new file window
When I search n huge file, I would like to extract all lines matching search (either full or part only) into a separate (new) file.
This is extremly useful feature observed in e.g. PSPad Editor

### Expected Behavior
After pressing button "List" in "Find" window, found lines should be placed inside new file.
Advanced feature: each line inside new file could be prefixed (on demand: checkbox) with original line number.
Example:
[ 1] Line with found text
[13] Another line with found text
[94] yet another line with found text

**Comment by alankilborn** [comment]
Right-click in _Search results_ area; you will get a popup menu like this:

![image](https://user-images.githubusercontent.com/77065706/157668088-7e99f398-1b54-41e8-9a21-e2ee53c90fef.png)

Choose _Select all_.

Invoke the menu again.

Choose _Copy_.

Create a new N++ file tab; paste there.

**Comment by alankilborn** [comment]
>After pressing button "List" in "Find" window

Is there a button with caption "List"?
Are you using a non-English localization in Notepad++ that has this?

**Comment by CennoxX** [comment]
> > After pressing button "List" in "Find" window
> 
> Is there a button with caption "List"? Are you using a non-English localization in Notepad++ that has this?

This was his expected behavior, with a new button named "List". As you showed, his problem can already be solved without addition of said button.
To also add line numbers use "Select all" (ctrl+a), unselect the first two lines (shift+down) and "Copy" (ctrl+c).

**Comment by alankilborn** [comment]
@CennoxX 

>This was his expected behavior, with a new button named "List"

Ah, OK, I understand now.

>To also add line numbers use "Select all" (ctrl+a), unselect the first two lines (shift+down) and "Copy" (ctrl+c).

Well, that only works well if you have exactly ONE search in your _Search results_ window, and only ONE file with hits.

A better technique is to copy everything and then use a regular expression replacement operation to remove unwanted search and pathname information.

**Comment by gawkla** [comment]
Hi all. Maybe not exact this topic, but there are two issues with 'search results' window:
1) sometimes (my case) I would NOT to have prefix 'line xxx:' before each line  [not possible now]
   why? because formating is wrong (line 1,    line 9999: it makes shifts between result lines)
   because this is waste of space if I want only to filter big log file for only certain lines
   
 2) search should behave like grep command: if searched sentence exists twice or more in one line this line should be presented once in results. Now each occurence provide one line in output, which easy give double line results, triple line results etc...

**Comment by alankilborn** [comment]
Solution for 1:  Right-click, _Copy Selected Line(s)

Solution for 2: Do search in Regular expression search mode, put `.*` at the end of your _Find what_ string

**Comment by gawkla** [comment]
Hi @alankilborn 
I know these workarounds but...
I am using notepad++ for log analyzing. Very often I have a lot of search results and have to see things directly in search result window. Seeing "line xx" as prefix, seeing shifted texts [because prefixed by different prefixes], seeing the same lines twice or more -> this is what is hurting me.
It is no problem for me to copy test to other window and clear everything. But it is wasting a lot of my time ;)

Why I have to focus on search result window? Because here I have filtered lines. If I hit interesting place I can double click it and see wide context in upper part of the window. Coping text to other window breake this feature.

My intention is not to force everybody to do things like that or not. But it would be great if we have possibility to customize things [for example disabling prefix].
Double lines in result window are like BUG for me and was reported by me in the past, but answer was "working as designed".

**Comment by alankilborn** [comment]
>seeing shifted texts [because prefixed by different prefixes]

This part has been fixed by a recent commit.

**Comment by gawkla** [comment]
> This part has been fixed by a recent commit.

Great news. I will check. What about other "issues" ?
They was reported by me many years ago. Should be reported again or author take it into consideration???
to summary:
- search results lines should be presented once (like grep output)
- there should be possibility to completely hide prefix: "line    xxx: " in search results (complete no prefix)

**Comment by alankilborn** [comment]
>What about other "issues" ?

Well, you kind of hijacked the original intent of THIS issue; the others that you refer to best belong in issues dedicated to them.

>They was reported by me many years ago.

If author declined them, there's not much you can do.  Author does change his mind, but is it likely?

>search results lines should be presented once (like grep output)

When double-clicking the search results lines, focus jumps to the editor window with the caret on the match.  With multiple output lines for a single source line in search results, this works well.  If a single line in search results contains many hits, what is your proposal for jumping to the Nth hit (N > 1)?

>there should be possibility to completely hide prefix: "line xxx: " in search results (complete no prefix)

Not sure I fully see the value in this, can you explain the real benefit, in better terms than "this is what is hurting me"?
